Source/WarpX.cpp

```diff
@@ -669,25 +683,33 @@ WarpX::ReadParameters ()
 
         pp.query("current_correction", current_correction);
         pp.query("v_galilean", m_v_galilean);
+        pp.query("v_comoving", m_v_comoving);
         pp.query("do_time_averaging", fft_do_time_averaging);
 
       // Scale the velocity by the speed of light
         for (int i=0; i<3; i++) m_v_galilean[i] *= PhysConst::c;
+        for (int i=0; i<3; i++) m_v_comoving[i] *= PhysConst::c;
 
 #   ifdef WARPX_DIM_RZ
         update_with_rho = true;  // Must be true for RZ PSATD
 #   else
-        if (m_v_galilean[0] == 0. && m_v_galilean[1] == 0. && m_v_galilean[2] == 0.) {
+        if (m_v_galilean[0] == 0. && m_v_galilean[1] == 0. && m_v_galilean[2] == 0. &&
+            m_v_comoving[0] == 0. && m_v_comoving[1] == 0. && m_v_comoving[2] == 0.) {
             update_with_rho = false; // standard PSATD
         }
         else {
-            update_with_rho = true;  // Galilean PSATD
+            update_with_rho = true;  // Galilean PSATD or comoving PSATD
         }
 #   endif
 
         // Overwrite update_with_rho with value set in input file
         pp.query("update_with_rho", update_with_rho);
 
+        if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.) {
+            AMREX_ALWAYS_ASSERT_WITH_MESSAGE(update_with_rho,
+                "psatd.update_with_rho must be equal to 1 for comoving PSATD");
+        }
```

Comment:
Note: the rho-free formulation of the comoving PSATD scheme is not yet implemented and I do not plan to do so in this PR.